Basic/ClientDriven/Assets/Scripts/NetworkObjectSpawner.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using Unity.Netcode;
+using UnityEngine;
+using UnityEngine.Networking;
+using Random = System.Random;
+
+namespace Unity.Multiplayer.Samples.ClientDriven
+{
+    /// <summary>
+    /// Spawn a NetworkObject at this transform's position when NetworkManer's server is started.
+    /// </summary>
+    /// <remarks>
+    /// A NetworkManager is expected to be part of the scene that this NetworkObject is a part of.
+    /// </remarks>
+    internal class NetworkObjectSpawner : MonoBehaviour
+    {
+        [SerializeField]
+        private NetworkObject prefabReference;
+        [SerializeField]
+        private NetworkManager m_NetworkManager;
```

Comment:
If I understood the resolution correctly, we don't need a scene reference anymore, since our assumption for this class is that a NetworkManager exists as part of this scene, and now knowing that NetworkManager.Singleton will be set by this behaviour's Start method.

Reply:
ok this version creates null references so I am gonna keep the NetworkManager reference

```
NullReferenceException: Object reference not set to an instance of an object
Unity.Multiplayer.Samples.ClientDriven.NetworkObjectSpawner.OnDestroy () (at Assets/Scripts/NetworkObjectSpawner.cs:32)
```
(Occurs if I use: `NetworkManager.Singleton.OnServerStarted -= OnServerStartedIngredientSpawn;` in OnDestroy() method)

Reply:
This just means that NetworkManager.Singleton has been dereferenced (likely that MonoBehaviour has been destroyed).

You can just check if NetworkManager.Singleton is valid before deregistering to the callback.

Reply:
Right ok I add it again. adding an extra line